### DISCUSSION TITLE
mpremote edit several files

### DISCUSSION THREAD

**Discussion by villeneuve** [micropython/micropython#11435] [General]
Hi!

First thing I've to say: big thanks for this v1.20 and for the new WebREPL now available on the Pico W :-)
mpremote edit file1 file2 etc..can edit several files 
However the mpremote behavior is to edit file1, save, close the editor, then edit file2 etc..
It can be convenient to have each file in an editor tab (when using an editor in a GUI like geany)
I have slightly modified the edit command in the file command.py (located in /usr/local/lib/python3.7/dist-packages/mpremote/ in my Raspberry Pi OS) to do so.
Here is my modification of the do-edit function (line 164):
```py
def do_edit(state, args):
    state.ensure_raw_repl()
    state.did_action()

    dest_all = []
    src_all = []
    if not os.getenv("EDITOR"):
        raise pyboard.PyboardError("edit: $EDITOR not set")
    for src in args.files:
        src = src.lstrip(":")
        dest_fd, dest = tempfile.mkstemp(suffix=os.path.basename(src))
        os.close(dest_fd)
        state.pyb.fs_touch(src)
        state.pyb.fs_get(src, dest, progress_callback=show_progress_bar)
        dest_all.append(dest)
        src_all.append(src)
    editor_arg = ' '.join(dest_all)
    print("edit", ' '.join(src_all))
    try:
        if os.system('%s %s' % (os.getenv("EDITOR"), editor_arg)) == 0:
            for i in range(len(src_all)):
                state.pyb.fs_put(dest_all[i], src_all[i], progress_callback=show_progress_bar)
    finally:
        for i in range(len(dest_all)):
            os.unlink(dest_all[i])
```
It's worth to note that the files are actually written to the board when the editor with all its tabs is closed. This can be tricky as it isn't the same when editing files on the PC itself.

..

**Comment by Josverl** [comment]
That would be nice, although the line

> `os.system('%s %s' % (os.getenv("EDITOR"), editor_arg)) == 0:`

Will not work on windows if there are spaces in EDITOR or the temp file names

**Reply by villeneuve** [reply]
> Will not work on windows if there are spaces in EDITOR or the temp file names

Ah! Too bad.
I can't do any test, I don't have any Windows machine here.
May be a test with `sys.platform` ?..
I succesfully tested it on GNU/Linux with 3 editors: geany, mu and nano.
